classes/manager.php

```diff
@@ -86,16 +95,58 @@ public static function init(): void {
 
         // Call self::saveprofile whenever the logs get flushed.
         $spool = function(\ExcimerLog $log) use ($started) {
-            profile::save($log, $started);
+            manager::onflush($log, $started);
         };
         $prof->setFlushCallback($spool, self::EXCIMER_LOG_LIMIT);
-        $prof->start();
 
         // Stop the profiler as a part of the shutdown sequence.
         \core_shutdown_manager::register_function(
             function() use ($prof) {
                 $prof->stop(); $prof->flush();
             }
         );
+
+        $prof->start();
+    }
+
+    /**
+     * Called when the Excimer log flushes.
+     *
+     * @param \ExcimerLog $log
+     * @param float $started
+     * @throws \dml_exception
+     */
+    public static function onflush(\ExcimerLog $log, float $started): void {
+        $stopped = microtime(true);
+        $duration = $stopped - $started;
+
+        if (self::isflagset(self::MANUAL_PARAM_NAME)) {
+            $logmethod = self::LOGMETHOD_MANUAL;
+            $dowesave = true;
+        } else {
+            $logmethod = (int) get_config('tool_excimer', 'excimerlogmethod');
+            $dowesave = ($duration * 1000) >= (int) get_config('tool_excimer', 'excimertrigger_ms');
+            if ($dowesave) {
+                $numrecorded = profile::getnumautoprofiles();
+                if ($numrecorded >= (int) get_config('tool_excimer', 'excimernum_slowest')) {
+                    if ($duration <= profile::getfastestautoprofile()->duration) {
+                        $dowesave = false;
+                    } else {
+                        profile::purgefastestautoprofile();
+                    }
+                }
+            }
+        }
+        if ($dowesave) {
+            profile::save($log, $logmethod, (int) $started, $duration);
+        }
+    }
+
+    public static function onnumslowsettingchange(string $name) {
```

Comment:
while vars can't have _ a function name can to be more readable

Reply:
This has been largely implemented.